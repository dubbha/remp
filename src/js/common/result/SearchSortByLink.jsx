import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export default class SearchSortByLink extends Component {
  handleClick = () => {
    const { onClickWithText, text } = this.props;

    if (onClickWithText) {
      onClickWithText(text);
    }
  }

  render() {
    const {
      text,
      active,
    } = this.props;

    return (
      <a
        role="link"
        tabIndex={0}
        className={classNames(
          'searchResult__link',
          { searchResult__link_active: active },
        )}
        onClick={this.handleClick}
      >
        {`${text}`}
      </a>
    );
  }
}

SearchSortByLink.defaultProps = {
  active: false,
  onClickWithText: null,
};

SearchSortByLink.propTypes = {
  text: PropTypes.string.isRequired,
  active: PropTypes.bool,
  onClickWithText: PropTypes.func,
};
