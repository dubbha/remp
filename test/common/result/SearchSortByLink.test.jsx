import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchSortByLink from 'common/result/SearchSortByLink';

describe('SearchSortByLink', () => {
  let props;

  beforeEach(() => {
    props = {
      text: 'TEXT',
      active: false,
      onClickWithText: jest.fn(),
    };
  });

  it('should render successfully', () => {
    const wrapper = shallow(<SearchSortByLink {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render active class for the active link', () => {
    const wrapper = shallow(<SearchSortByLink {...props} active />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should handle click if on click function is passed', () => {
    const wrapper = shallow(<SearchSortByLink {...props} />);

    wrapper.find('a').simulate('click');

    expect(props.onClickWithText).toBeCalled();
  });

  it('should silently ignore the click otherwise', () => {
    props.onClickWithText = null;
    const wrapper = shallow(<SearchSortByLink {...props} />);

    // Wrap in a function, otherwise the error will not be caught and the assertion will fail
    const wrapFn = () => wrapper.find('a').simulate('click');

    expect(wrapFn).not.toThrow();
  });
});
