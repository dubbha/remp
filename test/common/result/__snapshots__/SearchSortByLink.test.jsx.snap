// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSortByLink should render active class for the active link 1`] = `
<a
  className="searchResult__link searchResult__link_active"
  onClick={[Function]}
  role="link"
  tabIndex={0}
>
  TEXT
</a>
`;

exports[`SearchSortByLink should render successfully 1`] = `
<a
  className="searchResult__link"
  onClick={[Function]}
  role="link"
  tabIndex={0}
>
  TEXT
</a>
`;
